pkg/volume/csi/csi_attacher.go

```diff
@@ -102,17 +102,11 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		glog.V(4).Info(log("attachment [%v] for volume [%v] created successfully", attachID, csiSource.VolumeHandle))
 	}
 
-	// probe for attachment update here
-	// NOTE: any error from waiting for attachment is logged only.  This is because
-	// the primary intent of the enclosing method is to create VolumeAttachment.
-	// DONOT return that error here as it is mitigated in attacher.WaitForAttach.
-	volAttachmentOK := true
 	if _, err := c.waitForVolumeAttachment(csiSource.VolumeHandle, attachID, csiTimeout); err != nil {
```

Comment:
btw I noticed that CSI has default attachment time of 15s. That seems awfully short - is there anything in CSI that can return how much expected waiting time is going to be?

Reply:
No, there is no way how CSI driver could return estimated time.

On the other hand, attachment is handled by external attacher. External attacher still tries to attach the volume even when A/D controller timed out. A/D controller should periodically check (with exp. backoff?) if the volume got attached and does not need to wait in a blocking goroutine.